metals/src/main/scala/scala/meta/internal/metals/scalacli/ScalaCli.scala

```diff
@@ -227,34 +228,62 @@ class ScalaCli(
           .filter(requireMinVersion(_, minVersion))
           .map(p => Seq(p.toString))
       }
-      .getOrElse {
-        scribe.warn(
-          s"scala-cli >= $minVersion not found in PATH, fetching and starting a JVM-based Scala CLI"
-        )
-        val cp = ScalaCli.scalaCliClassPath()
-        Seq(
-          ScalaCli.javaCommand,
-          "-cp",
-          cp.mkString(File.pathSeparator),
-          ScalaCli.scalaCliMainClass,
-        )
-      }
-    cliCommand ++ Seq("bsp")
+  }
+
+  private lazy val cliCommand = {
+    localScalaCli.getOrElse {
+      scribe.warn(
+        s"scala-cli >= $minVersion not found in PATH, fetching and starting a JVM-based Scala CLI"
+      )
+      jvmBased()
+    }
+  }
+
+  def jvmBased(): Seq[String] = {
+    val cp = ScalaCli.scalaCliClassPath()
+    Seq(
+      ScalaCli.javaCommand,
+      "-cp",
+      cp.mkString(File.pathSeparator),
+      ScalaCli.scalaCliMainClass,
+    )
   }
 
   def loaded(path: AbsolutePath): Boolean =
     ifConnectedOrElse(st =>
       st.path == path || path.toNIO.startsWith(st.path.toNIO)
     )(false)
 
+  def setupIDE(path: AbsolutePath): Future[Unit] = {
+    localScalaCli
+      .map { cliCommand =>
+        val command = cliCommand ++ Seq("setup-ide", path.toString())
```

Comment:
Should we maybe run it for the current workspace? If a user adds new files then those will not be served by Scala CLI

Reply:
I do run it for the workspace folder's root.

Reply:
Ach, I missed that, thanks for explaining!